src/handler.ts

```diff
@@ -13,50 +13,44 @@ export const handler: Handler = async (event: SQSEvent): Promise<SQSBatchRespons
     logger.error('ERROR: event is not defined.');
     throw new Error('Event is empty');
   }
-
-  const altPromiseArray = event.Records.map((sqsRecord) => {
-    const request = JSON.parse(sqsRecord.body) as Request;
-
-    const document: DocumentModel = getDocumentFromRequest(request);
-    return generateAndUpload(document, request);
-  });
-
-  const results = await Promise.allSettled(altPromiseArray);
-  const ids = results
-    .map((result, index) => (result.status === 'fulfilled' ? null : event.Records[index].messageId))
-    .filter((item) => item !== null);
-  return {
-    batchItemFailures: ids.map((id) => ({ itemIdentifier: id })),
-  };
+  const batchItemFailures: SQSBatchItemFailure[] = [];
+  // eslint-disable-next-line no-restricted-syntax
+  for (const sqsRecord of event.Records) {
+    try {
+      const request = JSON.parse(sqsRecord.body) as Request;
+      const document: DocumentModel = getDocumentFromRequest(request);
+      // eslint-disable-next-line no-await-in-loop
+      await generateAndUpload(document, request);
+    } catch (error) {
+      console.error(error);
+      batchItemFailures.push({ itemIdentifier: sqsRecord.messageId });
+    }
+  }
+  return { batchItemFailures };
 };
 
 const generateAndUpload = async (document: DocumentModel, request: Request) => {
-  try {
-    logger.info(`Starting lambda to lambda invoke (data): ${JSON.stringify(document)}`);
-    const response = await invokePdfGenLambda(document, request.documentName);
-    logger.info('Finished lambda to lambda invoke, checking response');
+  logger.info(`Starting lambda to lambda invoke (data): ${JSON.stringify(document)}`);
+  const response = await invokePdfGenLambda(document, request.documentName);
+  logger.info('Finished lambda to lambda invoke, checking response');
 
-    if (response.StatusCode !== 200) {
-      logger.error(`Error invoking doc gen (lambda call failed with ${response.StatusCode}`);
-      throw new Error(`Error invoking doc gen (lambda call failed with ${response.StatusCode}`);
-    }
-    const responseString: string = new TextDecoder().decode(response.Payload);
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const responseJson: any = JSON.parse(responseString);
-    if (responseJson.statusCode !== 200) {
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      throw new Error(`Error returned from doc gen (${responseJson.statusCode}): ${responseJson.body}`);
-    }
+  if (response.StatusCode !== 200) {
+    logger.error(`Error invoking doc gen (lambda call failed with ${response.StatusCode}`);
```

Comment:
Now that we're logging out the catch block error in the handler, I don't think that we need to log the error here. We can just throw it because; the catch block in generate upload is logging it already, then throwing it out into the catch block in the handler to be logged again which will result in it being logged twice